mipdb/usecases.py

```diff
@@ -516,11 +435,111 @@ def _get_dataset_id(self, code, schema_id, conn):
 
 @emitter.handle("disable_dataset")
 def update_actions_on_dataset_disablement(record, conn):
+    update_actions(record, "DISABLE DATASET", conn)
+
+
+class TagSchema(UseCase):
+    def __init__(self, db: DataBase) -> None:
+        self.db = db
+
+    def execute(self, code, version, tag, key_value, add, remove) -> None:
+        if not tag and not key_value:
+            raise UserInputError("You need to provide a tag or/and a key value pair")
+        if add == remove:
+            raise UserInputError("You need to add or remove")
+
+        action = Action.ADD if add else Action.REMOVE
+        metadata = Schema(METADATA_SCHEMA)
+        schemas_table = SchemasTable(schema=metadata)
+
+        with self.db.begin() as conn:
+            schema_id = self._get_schema_id(code, version, conn)
+            properties = Properties(
+                schemas_table.get_schema_properties(schema_id, conn)
+            )
+            properties.update_properties(tag, key_value, action)
+            schemas_table.set_schema_properties(properties.properties, schema_id, conn)
+            action = "REMOVE SCHEMA TAG" if action == Action.REMOVE else "ADD SCHEMA TAG"
+
+            record = dict(
+                code=code,
+                version=version,
+                schema_id=schema_id,
+                action=action,
+            )
+            emitter.emit("tag_schema", record, conn)
+
+    def _get_schema_id(self, code, version, conn):
+        metadata = Schema(METADATA_SCHEMA)
+        schemas_table = SchemasTable(schema=metadata)
+        schema_id = schemas_table.get_schema_id(code, version, conn)
+        return schema_id
+
+
+@emitter.handle("tag_schema")
+def update_actions_on_schema_tagging(record, conn):
+    update_actions(record, record["action"], conn)
+
+
+class TagDataset(UseCase):
```

Comment:
The same remark as before. It's 4 use cases:
```
TagDataset
UntagDataset
AddPropertyToDataset
RemovePropertyFromDataset
```
Or if you do the "key=value" trick, it's 2 usecases `TagDataset`, `UntagDataset`.